### DISCUSSION TITLE
Incorrect cosine distance results

### DISCUSSION THREAD

**Discussion by Zeptorazer** [milvus-io/milvus#27327] [Q&A and General discussion]
Hi,

I am using Milvus 2.3 and encountering issues with the cosine similarity metric. In my vector database, I store 1024-dimensional LabelEmbeddings. Whenever I perform a vector search using the cosine metric, I receive results greater than 1. This problem persists across various vector indices, including FLAT, IVF_FLAT, and IVF_SQ8. However, when I use a different distance metric like IP, I obtain correct results. The issue remains consistent regardless of the number of entities in the database. In the code below I have loaded the DB with 9 embeddings, but the same problem occurred with 8000 embeddings as well.

Search and index parameters:
`index_params = {
        "metric_type":"COSINE",
        "index_type":"FLAT"
        }

        search_params = {
        "metric_type": "COSINE", 
        "offset": 0, 
        "ignore_growing": False
        }`

Collection schema:
      dbid =FieldSchema(
            name="dbid",
            dtype=DataType.INT64,
            is_primary=True,
        )
        nugget_id = FieldSchema(
            name="nugget_id",
            dtype=DataType.INT64,
        )
        document_id = FieldSchema(
            name="document_id",
            dtype=DataType.INT64
        )
        embedding_value = FieldSchema( 
            name="embedding_value",
            dtype=DataType.FLOAT_VECTOR,
            dim=1024,
        )
        embbeding_schema = CollectionSchema(
            fields=[dbid,nugget_id, document_id, embedding_value],
            description="Schema for nuggets",
            enable_dynamic_field=True,
        )`
        
Vector search:
        `results = adjusted_embedding_collection.search(
            data=[attribute_embedding], 
            anns_field="embedding_value", 
            param=self._search_params,
            limit=1000,
            expr= None,
            output_fields=['id','document_id'],
            consistency_level="Strong"
        )`
        
Results of my vector search:
Numpy cosine distance for nugget: 0; document: 0; distance: 0.4073899984359741
vector database cosine distance for nugget: 0; document: 0; distance: 11.583507537841797
Numpy cosine distance for nugget: 1; document: 0; distance: 0.4073899984359741
vector database cosine distance for nugget: 1; document: 0; distance: 11.58350658416748
Numpy cosine distance for nugget: 2; document: 0; distance: 0.6218472719192505
vector database cosine distance for nugget: 2; document: 0; distance: 7.264456748962402
Numpy cosine distance for nugget: 0; document: 1; distance: 0.4073899984359741
vector database cosine distance for nugget: 0; document: 1; distance: 11.58350658416748
Numpy cosine distance for nugget: 1; document: 1; distance: 0.4073899984359741
vector database cosine distance for nugget: 1; document: 1; distance: 11.58350658416748
Numpy cosine distance for nugget: 2; document: 1; distance: 0.6218472719192505
vector database cosine distance for nugget: 2; document: 1; distance: 7.264456748962402
Numpy cosine distance for nugget: 0; document: 2; distance: 0.4073899984359741
vector database cosine distance for nugget: 0; document: 2; distance: 11.58350658416748
Numpy cosine distance for nugget: 1; document: 2; distance: 0.4073899984359741
vector database cosine distance for nugget: 1; document: 2; distance: 11.58350658416748
Numpy cosine distance for nugget: 2; document: 2; distance: 0.6218472719192505
vector database cosine distance for nugget: 2; document: 2; distance: 7.264456748962402

Does a solution or some form of workaround exist for this problem?

Thank you very much!

**Comment by xiaofan-luan** [comment]
the new 2.3.1 should solve the problem.
Please test on it, thanks
